### ISSUE TITLE
Create a prover process

### ISSUE THREAD

**Issue by cmwaters** [celestiaorg/celestia-zkevm-ibc-demo#6]
We need a separately running process that can take as inputs the JSON RPC of both the reth and Celestia nodes and have a simple public API that generates groth16 proofs (and the accompanying public inputs) given a specified height range. This would then be used by the relayer to update simapps ZK IBC Client that tracks the state of the reth rollup.

FYI we have a similar prover process going the other direction known as the `operator` which generates succinct proofs of the tendermint header to be processed by the reth node.

**Comment by cmwaters** [comment]
Ref: https://github.com/S1nus/blevm/tree/main

**Comment by rootulp** [comment]
@S1nus is this issue close-able given https://github.com/S1nus/blevm exists? 

There is a separate issue to wrap blevm in a GRPC service: https://github.com/celestiaorg/celestia-zkevm-ibc-demo/issues/41.

**Comment by S1nus** [comment]
> @S1nus is this issue close-able given https://github.com/S1nus/blevm exists? 
> 
> 
> 
> There is a separate issue to wrap blevm in a GRPC service: https://github.com/celestiaorg/celestia-zkevm-ibc-demo/issues/41.

Yes the blevm repo can house the wrapping service + proof programs

**Comment by rootulp** [comment]
https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/56 exists and targets this repo so I think the intention was for the wrapping service to live in this repo in the [provers](https://github.com/celestiaorg/celestia-zkevm-ibc-demo/tree/main/provers) directory.

IMO it's not a big deal if it lives in a separate repo either but worth clarifying with @tuxcanfly. I'll close this issue as resolved for now then.
